### PR TITLE
advanced-rest-client: init at 17.0.9

### PR DESCRIPTION
**Pull request by kashw2**
## Description of changes

https://www.advancedrestclient.com/
https://github.com/advanced-rest-client/arc-electron

Advanced Rest Client (ARC) is an application made for testing web APIs' similar to Postman.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kashw2** [comment]
Result of `nixpkgs-review pr 247651` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>arc</li>
  </ul>
</details>

File: pkgs/by-name/ad/advanced-rest-client/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  src = fetchurl {
+    url = "https://github.com/advanced-rest-client/arc-electron/releases/download/v${version}/arc-linux-${version}-x86_64.AppImage";
+    hash = "sha256-BGHrhv+tfuczlkpSfBuF4GFURU9lVjHXvvXqWxEVd9U=";
+  };
+
+  extraInstallCommands = ''
+    mkdir -p $out/share
+    cp -rt $out/share ${desktopItem}/share/applications
+    mv $out/bin/${pname}-${version} $out/bin/${pname}
```
Comment (pbsds) [inline pkgs/by-name/ad/advanced-rest-client/package.nix:23]:
```suggestion
```

No longer needed as of nixos 24.05

**Review by pbsds — COMMENTED** [review]
Result of `nixpkgs-review pr 247651` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>advanced-rest-client</li>
  </ul>
</details>

Error is:

```
mv: cannot stat '/nix/store/r354s67d5p5v9vkr7hrk6fr4bwqmhvk8-advanced-rest-client-17.0.9/bin/advanced-rest-client-17.0.9': No such file or directory
```

File: pkgs/by-name/ad/advanced-rest-client/package.nix (hunk 1 above)
Comment (kashw2) [reply pkgs/by-name/ad/advanced-rest-client/package.nix]: fixed in https://github.com/NixOS/nixpkgs/pull/247651/commits/302ba76deeebc18263fb000293af2a4106c3449f

**Review by pbsds — COMMENTED** [review]
It seems the extracted appimage has a bunch of icons. Maybe refer to `nixpkgs/pkgs/by-name/an/anytype/package.nix` on how to add them to the output

File: pkgs/by-name/ad/advanced-rest-client/package.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  src = fetchurl {
+    url = "https://github.com/advanced-rest-client/arc-electron/releases/download/v${version}/arc-linux-${version}-x86_64.AppImage";
+    hash = "sha256-BGHrhv+tfuczlkpSfBuF4GFURU9lVjHXvvXqWxEVd9U=";
+  };
+
+  appimageContents = appimageTools.extractType2 { inherit pname version src; };
+
+  extraInstallCommands = ''
+    mkdir -p $out/share
+    cp -rt $out/share ${desktopItem}/share/applications
```
Comment (SuperSandro2000) [inline pkgs/by-name/ad/advanced-rest-client/package.nix:24]: Use copyDekstopItems for this instead

File: pkgs/by-name/ad/advanced-rest-client/package.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    chmod -R +w $out/share
+    install -m 444 -D ${appimageContents}/${pname}.desktop -t $out/share/applications
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=AppRun' 'Exec=${pname}'
```
Comment (SuperSandro2000) [inline pkgs/by-name/ad/advanced-rest-client/package.nix:28]:
```suggestion
    install -m 444 -D ${appimageContents}/advanced-rest-client.desktop -t $out/share/applications
    substituteInPlace $out/share/applications/advanced-rest-client.desktop \
      --replace 'Exec=AppRun' 'Exec=advanced-rest-client'
```
using pname like this is an anti pattern

**Comment by SuperSandro2000** [comment]
Do we really want to add a new software to nixpkgs which is already EOL upstream and unmaintained since 2 years?

**Comment by kashw2** [comment]
When this PR was originally made it hadn't been nearly as unmaintained and there were branches that were still active. I'd opted to include it as part of my flake's packages after a bit until migrating to Bruno not long after so I'm not too concerned about this PR anymore

**Pull request closed without merging**